projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts

```diff
@@ -223,4 +229,66 @@ export class HsCompositionsMickaService {
     }
     return params;
   }
+
+  /**
+   * Get information about the selected composition
+   * @param composition - Composition selected
+   
+   */
+  async getInfo(composition: HsMapCompositionDescriptor): Promise<any> {
+    const compLinks = composition.link || composition.links;
+    if (compLinks === undefined) {
+      return;
+    }
+    const compUrls = this.getCompositionUrls(compLinks);
+    let info: any = {};
+    let url = '';
+    Array.isArray(compUrls) ? (url = compUrls[0]) : (url = compUrls);
+    try {
+      info = await this.hsCompositionsParserService.loadInfo(url);
+      //TODO: find out if this is even available
+      // info.thumbnail = this.HsUtilsService.proxify(composition.thumbnail);
+      info.metadata = {
+        record_url:
+          composition.endpoint.url.replace('csw', 'record/basic/') +
+          composition.id,
+      };
+      return info;
+    } catch (e) {
+      this.hsToastService.createToastPopupMessage(
+        this.hsLanguageService.getTranslation(
+          'COMPOSITIONS.errorWhileLoadingCompositionMetadata',
+          undefined
+        ),
+        this.hsLanguageService.getTranslationIgnoreNonExisting(
+          'ERRORMESSAGES',
+          e.status ? e.status.toString() : e.message,
+          {url: url}
+        ),
+        {
+          disableLocalization: true,
+          serviceCalledFrom: 'HsCompositionsMickaService',
+        }
+      );
+    }
+  }
+  /**
+   * Get composition urls
+   * @param compData - Composition data
+   */
+  getCompositionUrls(compData: any): string | Array<string> {
+    if (typeof compData == 'string') {
+      return compData;
+    }
+    if (typeof compData == 'object' && compData.url !== undefined) {
+      return compData.url;
+    }
+    return compData.map((link) =>
+      typeof link == 'object' && link.url !== undefined ? link.url : link
+    );
+  }
+
+  delete(e: HsEndpoint, c: HsMapCompositionDescriptor) {
+    console.warn('Delete method for Micka compositions not implemented', e, c);
+  }
```

Comment:
Hmm can cant remember what I ment by this.  But thinking about this now I guess user is not supposed to remove micka records directly, right @fzadrazil ?

Reply:
There are two cases:

1. Layman records with metadata in Micka - HSLayers should primarily display Layman record, not Micka, once the Layman record is deleted, it is automatically deleted also in Micka (Layman takes care of that)
2. Micka records with no link to Layman - user should not delete those in HSLayers

In case of some error when Layman record is removed but not Micka record, it is still responsibility of Layman to remove this, not HSLayers.

Reply:
Deleting Micka records from HSLayers is already prevented in the UI. By setting `editable = false` property, the button is hidden here https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.html#L20 and further prevented here https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.ts#L96

So the code is fine with me like this. I propose to add some comment to the delete() method itself to make it clear for future ourselves, why it is not implemented 🙂